cmd/kops-controller/controllers/gceipam.go

```diff
@@ -139,6 +147,46 @@ func (r *GCEIPAMReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		}
 	}
 
+	if len(node.Spec.PodCIDRs) != 0 {
```

Comment:
you can make this simpler , this field is already validated in the apiserver https://github.com/kubernetes/kubernetes/blob/54d42e6a658f84ea66593d3107f41db6931a5fbf/pkg/apis/core/validation/validation.go#L4247-L4281

```suggestion
       netutils "k8s.io/utils/net" 


	if len(node.Spec.PodCIDRs) != 1 {
	    return ctrl.Result{}, nil
	 }
	 if netutils.IsIPv4CIDR(node.Spec.PodCIDRs[0]) {
	      return ctrl.Result{}, nil
	 }
	 ...
	 patch condition here to unset NetworkUnavailable
	 ...
	    
```